### PR TITLE
openstack-deployment: diff installed packages with old job runs

### PR DESCRIPTION
**Pull request by djz88**
This PR enhances diff of installed packages by doing diff of old job runs.

### CODE REVIEW

**Comment by djz88** [comment]
Check artifacts in rpms/diffs.
Crowbar CI: https://ci.suse.de/job/openstack-crowbar/995/
Ardana CI: https://ci.suse.de/job/openstack-ardana/772/

File: scripts/jenkins/cloud/ansible/roles/list_packages/defaults/main.yml [hunk 1]
```diff
@@ -20,4 +20,12 @@ state1: "after_deploy"
 state2: "after_update"
 effective_user: "{{ (cloud_product == 'ardana') | ternary('ardana','root') }}"
 local_tmp_dir: "/tmp"
-diff_tmp_dir: "{{ lookup('env', 'WORKSPACE') | ternary (workspace_path, local_tmp_dir) }}" 
+diff_tmp_dir: "{{ lookup('env', 'WORKSPACE') | ternary (workspace_path, local_tmp_dir) }}"
+build_id: "{{ lookup('env', 'BUILD_ID') if lookup('env', 'BUILD_ID') else '764' }}"
```
Comment (flaviodsr) [inline scripts/jenkins/cloud/ansible/roles/list_packages/defaults/main.yml:24]: what's the reason for 764?

I think the same can reached using `{{ lookup('env', 'BUILD_ID') | default('764', true) }}`

File: scripts/jenkins/cloud/ansible/roles/list_packages/tasks/packages_from_old_jobs.yml [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+    file1: "{{ diff_tmp_dir }}/rpms/old_jobs/{{ item_virtual_hosts.jobNumber }}/{{ item_virtual_hosts.currentFilename}}"
+    file2: "{{ diff_tmp_dir }}/rpms/{{ item_virtual_hosts.currentFilename }}"
+    output: "{{ diff_tmp_dir }}/rpms/diffs/{{ item_virtual_hosts.jobNumber }}_{{item_virtual_hosts.currentFilename}}"
+  delegate_to: localhost
+  register: diff_files
+  loop: "{{ __all_items.combine_files }}"
+  loop_control:
+    loop_var: item_virtual_hosts
+  when: __all_items.combine_files is not none
+  #when: _(_all_items.combine_files is not null) OR _(_all_items.combine_files is not none) OR (__all_items.combine_files is defined)
```
Comment (flaviodsr) [inline scripts/jenkins/cloud/ansible/roles/list_packages/tasks/packages_from_old_jobs.yml:106]: any reason for leaving this comment here?

Comment (djz88) [reply scripts/jenkins/cloud/ansible/roles/list_packages/tasks/packages_from_old_jobs.yml]: no reason, just leftover, I will remove it Thx

File: scripts/jenkins/cloud/ansible/roles/list_packages/defaults/main.yml (hunk 1 above)
Comment (djz88) [reply scripts/jenkins/cloud/ansible/roles/list_packages/defaults/main.yml]: 784 value here is for manual deployment... just thinking if I should keep it there or not.

Comment (djz88) [reply scripts/jenkins/cloud/ansible/roles/list_packages/defaults/main.yml]: I reworked it. So now if not defined it uses empty default and it means no diffs of old runs in manual deployment (If you wanna use it just define number in vars)

Comment (flaviodsr) [reply scripts/jenkins/cloud/ansible/roles/list_packages/defaults/main.yml]: perfect, thank you!

**Pull request merged**